std/range/package.d

```diff
@@ -6590,11 +6595,53 @@ if (isForwardRange!RangeOfRanges &&
     }
 
     auto opSlice() { return this; }
+    
+    static if (hasSlicing!(ElementType!RangeOfRanges))
+    {
+        auto opSlice(int start, int end)
+        {
+            return _input.map!(a => a[start..end]).array.transposed;
+        }
+    }
 
 private:
     RangeOfRanges _input;
 }
 
+struct TransposedIndex(RangeOfRanges)
```

Comment:
That works with the supplied tests, but fails for `[[1,2,3],[4,5,6]].transposed[0][0] = 0;`. Admittedly, that same test fails for `[[1,2,3],[4,5,6]].transposed.front.front = 0;`, so maybe that's better?